lib/vtls/schannel.c

```diff
@@ -414,6 +414,59 @@ get_cert_location(TCHAR *path, DWORD *store_name, TCHAR **store_path,
 }
 #endif
 
+static CURLcode
+static void schannel_session_free(void *ptr);
+
+static CURLcode
+schannel_client_certificate(struct Curl_easy *data, struct connectdata *conn,
+                            int sockindex)
+{
+  struct ssl_connect_data *connssl = &conn->ssl[sockindex];
+  SECURITY_STATUS status = 0;
+  SecPkgContext_IssuerListInfoEx dist_names;
+  CURLcode result;
+  unsigned long flags = BACKEND->ret_flags;
+
+  infof(data, "schannel: a client certificate has been requested\n");
+
+  /* Retrieve the distinguished names */
+  status =
+    s_pSecFn->QueryContextAttributes(&BACKEND->ctxt->ctxt_handle,
+                                     SECPKG_ATTR_ISSUER_LIST_EX,
+                                     &dist_names);
+  if(status != SEC_E_OK) {
+    infof(data, "QueryContextAttributes failed. status = %d, %ul ", status,
+          flags);
+    return CURLE_SSL_CONNECT_ERROR;
+  }
+  else {
+    /* Tell the application we have distinguished names. It is likely that
+        the application will pause the easy handle if user interaction is
+        required. */
+    if(data->set.ssl.fdistnames) {
+      result = (*data->set.ssl.fdistnames)(&dist_names,
+                                           data->set.ssl.fdistnamesp);
+      if(result != CURL_SSL_DN_FUNCTION_OK) {
+        failf(data, "dist names callback failed. status = %d, %ul ", status,
+              flags);
+         s_pSecFn->FreeContextBuffer(dist_names.aIssuers);
+        return result;
```

Comment:
This return looks wrong. It'll return the callback return value which isn't CURLE_*

Reply:
Fixed it.